op-acceptance-tests: Revise safedb EL sync tests

When we 
```sh
➜  safeheaddb_elsync git:(develop) cd optimism/op-acceptance-tests/tests/safeheaddb_elsync
➜  safeheaddb_elsync git:(develop) grep "func Test" safeheaddb_test.go 
func TestTruncateDatabaseOnELResync(gt *testing.T) {
func TestNotTruncateDatabaseOnRestartWithExistingDatabase(gt *testing.T) {
➜  safeheaddb_elsync git:(develop) ls
init_test.go       safeheaddb_test.go
➜  safeheaddb_elsync git:(develop) go test ./... -count 1
```
It attempts to spawn a sysgo shared state, and runs all tests inside the directory, `TestTruncateDatabaseOnELResync`, `TestNotTruncateDatabaseOnRestartWithExistingDatabase`

Every test passes when we make the test share the state.

However, if we target single test to run:
```sh
➜  safeheaddb_elsync git:(develop)cd optimism/op-acceptance-tests/tests/safeheaddb_elsync
➜  safeheaddb_elsync git:(develop)go test ./... -run ^TestNotTruncateDatabaseOnRestartWithExistingDatabase$  -count=1
```
It fails. Example logs:
```
    assertions.go:369:                  Error Trace:    /Users/changwan/Documents/optimism/op-devstack/dsl/safedb.go:25
    assertions.go:369:                                                          /Users/changwan/Documents/optimism/op-devstack/dsl/l2_cl.go:335
    assertions.go:369:                                                          /Users/changwan/Documents/optimism/op-acceptance-tests/tests/safeheaddb_elsync/safeheaddb_test.go:64
    assertions.go:369:                  Error:          "4" is not less than or equal to "0"
    assertions.go:369:                  Test:           TestNotTruncateDatabaseOnRestartWithExistingDatabase
    assertions.go:369:                  Messages:       safe head db did not go back far enough
    assertions.go:369:   
```
Not flaky but consistent failure.

Need to revise this tests to pass even we run individually.

--- a/op-acceptance-tests/tests/safeheaddb_elsync/safeheaddb_test.go
+++ b/op-acceptance-tests/tests/safeheaddb_elsync/safeheaddb_test.go
@@ -42,14 +42,13 @@ func TestNotTruncateDatabaseOnRestartWithExistingDatabase(gt *testing.T) {
 	t := devtest.SerialT(gt)
 	sys := presets.NewSingleChainMultiNode(t)
 
-	startSafeBlock := sys.L2CLB.SafeL2BlockRef().Number
-
 	dsl.CheckAll(t,
 		sys.L2CL.AdvancedFn(types.LocalSafe, 1, 30),
 		sys.L2CLB.AdvancedFn(types.LocalSafe, 1, 30))
-
 	sys.L2CLB.Matched(sys.L2CL, types.LocalSafe, 30)
-	sys.L2CLB.VerifySafeHeadDatabaseMatches(sys.L2CL)
+
+	preRestartSafeBlock := sys.L2CLB.SafeL2BlockRef().Number
+	sys.L2CLB.VerifySafeHeadDatabaseMatches(sys.L2CL, dsl.WithMinRequiredL2Block(preRestartSafeBlock))
 
 	// Restart the verifier op-node, but not the EL so the existing chain data is not deleted.
 	sys.L2CLB.Stop()
@@ -61,5 +60,5 @@ func TestNotTruncateDatabaseOnRestartWithExistingDatabase(gt *testing.T) {
 	sys.L2CLB.Matched(sys.L2CL, types.LocalSafe, 30)
 	sys.L2CLB.Advanced(types.LocalSafe, 1, 30) // At least one safe head db update after resync
 
-	sys.L2CLB.VerifySafeHeadDatabaseMatches(sys.L2CL, dsl.WithMinRequiredL2Block(startSafeBlock))
+	sys.L2CLB.VerifySafeHeadDatabaseMatches(sys.L2CL, dsl.WithMinRequiredL2Block(preRestartSafeBlock))
 }

